_studio/mfx_lib/decode/vp9/src/mfx_vp9_dec_decode_hw.cpp

```diff
@@ -85,6 +87,141 @@ bool CheckHardwareSupport(VideoCORE *p_core, mfxVideoParam *p_video_param)
     return true;
 }
 
+class ReferenceFrameStorage
+{
+private:
+    class FrameLocker
+    {
+    private:
+        UMC::FrameMemID m_frameId = -1;
+        std::shared_ptr<mfx_UMC_FrameAllocator> m_frameAllocator;
+        bool ValidFrameId()
+        {
+            return (m_frameId >= 0);
+        }
+
+        void Lock()
+        {
+            if (ValidFrameId())
+            {
+                if (m_frameAllocator->IncreaseReference(m_frameId) != UMC::UMC_OK)
+                {
+                    throw vp9_exception(UMC::UMC_ERR_FAILED);
+                }
+            }
+        }
+
+        void Unlock()
+        {
+            if (ValidFrameId())
+            {
+                if (m_frameAllocator->DecreaseReference(m_frameId) != UMC::UMC_OK)
+                {
+                    throw vp9_exception(UMC::UMC_ERR_FAILED);
+                }
+            }
+        }
+
+        FrameLocker(const FrameLocker &) = delete;
+        FrameLocker(const FrameLocker &&) = delete;
```

Comment:
cnonical signature is
`FrameLocker(FrameLocker &&)`

Reply:
Done.